exporter/collector/spandata.go

```diff
@@ -149,6 +150,13 @@ func pdataAttributesToOTAttributes(attrs pcommon.Map, resource pcommon.Resource)
 				otAttrs = append(otAttrs, attribute.Int64(k, v.Int()))
 			case pcommon.ValueTypeDouble:
 				otAttrs = append(otAttrs, attribute.Float64(k, v.Double()))
+			case pcommon.ValueTypeSlice:
```

Comment:
Seems like we should use json-encoded strings for these: https://opentelemetry.io/docs/specs/otel/common/#attribute

> For protocols that do not natively support non-string values, non-string values SHOULD be represented as JSON-encoded strings.

Seems like as a fallback, we could just use `v.AsString()`, as that handles this case as well as Map and Bytes: https://github.com/open-telemetry/opentelemetry-collector/blob/15f10456afcdc57288a56376bc087d3594599104/pdata/pcommon/value.go#L367

Reply:
Good catch - I'll take a look at that!

Reply:
@dashpole taking a look at your comment in more depth, I originally implemented this with a simple `v.AsString()` since that `json.Marshal`s under the hood anyways as you pointed out. I changed approaches to this one after I realized the behavior would still be different, as the collector exporter would include extra characters `[`  `]` `"`.

I think having the primary case do `json.Marshal(v.At(i).AsRaw())` is a reasonable approach.

Reply:
Are the extra characters problematic?  It seems like it would be good to follow the specification unless we have good reason to deviate

Reply:
tl;dr: Better UX through consistent experience + more intuitive interface, and a different interpretation of spec.

----
@dashpole I had chosen to move to this approach for two main reasons:
1. match the other GCP Python exporter's behavior to provide a consistent experience within Cloud Trace
2. As a user, when I want to apply filters from within Cloud Trace for Slice attributes, manually entering the extra chars from JSON serialization seems unintuitive.

My use case in particular is scraping HTTP headers as attributes. Given a header `X-Custom-Header` with values `value1` and `value2`, I'd expect I could go into Trace Explorer and filter for `http.request.header.x_custom_header: value1,value2`. Having to filter for `http.request.header.x_custom_header: "[\"value1\", \"value2\"]"` feels unusual to me and would probably require further documentation.

I'm also not entirely convinced that part of spec applies in this case, as Cloud Trace does natively support non-string values as seen with the primitives. It just doesn't support the array of primitives, which is not the condition given in the original quoted part of spec.

>The attribute value is either:
>  - A primitive type: string, boolean, double precision floating point (IEEE 754-1985) or signed 64 bit integer.
>  - An array of primitive type values. The array MUST be homogeneous, i.e., it MUST NOT contain values of different types.

>For protocols that do not natively support non-string values, non-string values SHOULD be represented as JSON-encoded strings.

Ultimately its your package, so I'm happy to implement how you'd like. Just let me know and I'll make whatever changes necessary!

Reply:
also, apologies for the verbosity. This is my first PR to a widely adopted open-source project, so I'm trying to be thorough, not pedantic :sweat_smile:

Reply:
No worries.  I'll raise this with the rest of my team, and get back to you.

Reply:
sounds good, thanks for your help!

Reply:
We decided to use the JSON-encoding for slices

Reply:
@dashpole sorry, I missed this notification. Sounds good, json-encoded it is! I'll update the implementation shortly. Thanks for looking into it.